runtime/ms_rest_azure/lib/ms_rest_azure/azure_service_client.rb

```diff
@@ -41,7 +41,7 @@ def get_long_running_operation_result(azure_response, custom_deserialization_blo
       polling_state = PollingState.new(azure_response, @long_running_operation_retry_timeout)
       request = azure_response.request
 
-      if !AsyncOperationStatus.is_terminal_status(polling_state.status)
+      while !AsyncOperationStatus.is_terminal_status(polling_state.status)
```

Comment:
Refer [.NET SDK Code](https://github.com/Azure/azure-sdk-for-net/blob/psSdkJson6/src/SdkCommon/ClientRuntime.Azure/ClientRuntime.Azure/AzureClientExtensions.cs#L107)  and [Node SDK Code](https://github.com/Azure/azure-sdk-for-node/blob/0a52678ea7b3a24a478975f7169fc30e9fc9759e/runtime/ms-rest-azure/lib/azureServiceClient.js#L119) for the reason of this change.